src/hotspot/cpu/s390/stubGenerator_s390.cpp

```diff
@@ -2857,6 +2858,48 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  address generate_nmethod_entry_barrier() {
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "nmethod_entry_barrier");
+
+    address start = __ pc();
+
+    // Save caller's sp & return_pc
+    __ push_frame(frame::z_abi_16_size);
+    __ z_stmg(Z_R14, Z_R15, _z_abi16(callers_sp), Z_SP);
```

Comment:
Please use `save_return_pc()`. Z_R15 = Z_SP is already saved by push_frame.

Reply:
You should not rely on Z_R14 being the return register in all circumstances and for all future. There are comments in the Principles of Operation and elsewhere that Z_R7 might assume a similar role. With save_return_pc() you are on the safe side and the code speaks for itself (no knowledge about Z_R14 required).

Reply:
Makes sense. I've made this change.